### DISCUSSION TITLE
how can I wire a simple java socket connection through KeyCloak?

### DISCUSSION THREAD

**Discussion by tombrus** [keycloak/keycloak#16503] [Q&A]
We have a very simple client/server setup with text line based messages through a plain TCP socket, written in java, full duplex, long living connections. Similar to a chat app. The servers runs on a host and listens on a specific port, the client connects to that host/port combi.

One of our clients wants to secure the connection with KeyCloak, but KeyCloak is new to me.

_What is the best way to connect the server and client through KeyCloak in this setup?_

I can customize both client and server end code.
I am looking for the option with the least changes :-) 
I am hesitant to pull in lots of libraries... we use only plain java now.

thanks! Tom

**Comment by sschu** [comment]
What Keycloak does is exchange some form of client/user authentication for tokens in form of a JWT using a protocol called OAuth2. So I guess you would have to support passing a token to authorise users/clients as part of your protocol.
